File: frontendloja/src/services/CategoriaService.js
import axios from 'axios';

export class CategoriaService {
    URL = "http://localhost:8080/api/categoria";

    inserir(categoria) {
        return axios.post(this.URL, categoria);
    }

    alterar(categoria) {
        return axios.put(this.URL, categoria);
    }

    excluir(id) {
        return axios.delete(this.URL + "/" + id);
    }

    listar(page, size) {
        return axios.get(this.URL + "?page=" + page + "&size=" + size);
    }
}

File: back/src/main/java/com/LojaVirtual/services/CategoriaService.java
package com.LojaVirtual.services;

import java.util.NoSuchElementException;
import java.util.Date;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import com.LojaVirtual.entities.Categoria;
import com.LojaVirtual.repositories.CategoriaRepository;

@Service
public class CategoriaService {

    @Autowired
    private CategoriaRepository categoriaRepository;

    public Page<Categoria> buscarTodos(Pageable pageable) {
        return categoriaRepository.findAll(pageable);
    }

    public Categoria buscarPorId(Long id) {
        return categoriaRepository.findById(id).get();
    }

    public Categoria inserir(Categoria categoria) {
        categoria.setDataCriacao(new Date());
        Categoria categoriaNovo = categoriaRepository.saveAndFlush(categoria);
        return categoriaNovo;
    }

    public Categoria alterar(Categoria categoria) {
        categoria.setDataAtualizacao(new Date());
        return categoriaRepository.saveAndFlush(categoria);
    }

    public void excluir(Long id) {
        Categoria categoria = categoriaRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("Categoria não encontrada."));
        categoriaRepository.delete(categoria);
    }
}
